### PR TITLE
OAuth Improvements

### PR DESCRIPTION
**Pull request by dilallkx**
**CHANGES:**
- Version bumped to 1.5.7
- Provider-specific sessions & user database entries
- Timer to clean up stale/timed out OAuth login requests
- Error params on failed Oauth login
- UT updates

**TESTING:**
- Cypress & UT pass
- Manual verification of fixes

### CODE REVIEW

File: go-packages/meep-users/db.go [hunk 1]
```diff
@@ -196,11 +202,14 @@ func (pc *Connector) DeleteTable(tableName string) (err error) {
...
 // CreateUser - Create new user
-func (pc *Connector) CreateUser(username string, password string, role string, sboxname string) (err error) {
+func (pc *Connector) CreateUser(provider string, username string, password string, role string, sboxname string) (err error) {
 	// Validate input
 	if username == "" {
 		return errors.New("Missing username")
 	}
+	if username == "" {
+		return errors.New("Missing username")
+	}
```
Comment (roymx) [inline go-packages/meep-users/db.go:212]: Checking `username == ""` twice

File: go-packages/meep-users/db.go [hunk 2]
```diff
@@ -449,7 +484,7 @@ func (pc *Connector) AuthenticateUser(username string, password string) (authent
 // isValidRole - does role exist
 func isValidRole(role string) error {
 	switch role {
-	case RoleUser, RoleSuper:
+	case RoleUser, RoleAdmin:
 		return nil
 	}
 	return errors.New("Inalid role")
```
Comment (roymx) [inline go-packages/meep-users/db.go:490]: Typo `Inalid`

**Review by roymx — APPROVED** [review]
Please resolve 2 minor changes

File: go-packages/meep-users/db.go (hunk 1 above)
Comment (dilallkx) [reply go-packages/meep-users/db.go:212]: Good catch! Fixed.

File: go-packages/meep-users/db.go (hunk 2 above)
Comment (dilallkx) [reply go-packages/meep-users/db.go:490]: Fixed.

**Pull request merged**